### PR TITLE
Fix error querying PineconeVectorStore using sparse query mode

### PR DESCRIPTION
**Pull request by Javtor**
# Description

Pinecone always expects a vector (or technically also a record id) when querying. When using sparse query mode, we were setting the vector to None and making the request with that, resulting in an error. Changed to make the request with a vector filled with zeroes instead.

## Version Bump?

- [x] Yes

## Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

File: llama-index-integrations/vector_stores/llama-index-vector-stores-pinecone/llama_index/vector_stores/pinecone/base.py [hunk 1]
```diff
@@ -433,7 +433,7 @@ def query(self, query: VectorStoreQuery, **kwargs: Any) -> VectorStoreQueryResul
                     "values": [v * (1 - query.alpha) for v in sparse_vector["values"]],
                 }
 
-        query_embedding = None
+        query_embedding = [0.0] * len(query.query_embedding)
```
Comment (logan-markewich) [inline llama-index-integrations/vector_stores/llama-index-vector-stores-pinecone/llama_index/vector_stores/pinecone/base.py:436]: This is specifically to fix the case where query_mode is sparse right?

Won't providing a vector of zeros still kind of influence the search result? If we are in sparse mode, should we just query without the vector kwarg? Or should we set the alpha to completely ignore the Dense zero vector?

Comment (Javtor) [reply llama-index-integrations/vector_stores/llama-index-vector-stores-pinecone/llama_index/vector_stores/pinecone/base.py:436]: yup

The issue is that if we query without the vector kwarg, we get an error when making the request. In their hybrid search tutorial, pinecone says we should scale the dense vector by alpha (in this case zero) before querying, so using a vector of zeros in this case should be fine (it's overwritten later for other query modes anyways) https://www.pinecone.io/learn/hybrid-search-intro/

Comment (logan-markewich) [reply llama-index-integrations/vector_stores/llama-index-vector-stores-pinecone/llama_index/vector_stores/pinecone/base.py:436]: Ah great, that works for me!

**Review by logan-markewich — APPROVED** [review]

File: llama-index-integrations/vector_stores/llama-index-vector-stores-pinecone/llama_index/vector_stores/pinecone/base.py (hunk 1 above)
Comment (logan-markewich) [reply llama-index-integrations/vector_stores/llama-index-vector-stores-pinecone/llama_index/vector_stores/pinecone/base.py:436]: Ok wait, one more worry haha -- we should ensure that the query embedding is not none before doing this

Comment (Javtor) [reply llama-index-integrations/vector_stores/llama-index-vector-stores-pinecone/llama_index/vector_stores/pinecone/base.py]: true, added a check

**Pull request merged**